File: jApp/src/main/java/j/s/yarlykov/data/provider/ForecastProvider.java
package j.s.yarlykov.data.provider;

import java.io.Serializable;
import java.util.Arrays;
import java.util.Date;
import java.util.List;

import j.s.yarlykov.R;

public class ForecastProvider {

    private static ForecastProvider instance;
    public static ForecastProvider getInstance() {
        if (instance == null) {
            instance = new ForecastProvider();
        }
        return instance;
    }

    private List<Forecast> forecasts;
    private ForecastProvider() {
        forecasts = Arrays.asList(
                new Forecast(R.drawable.rain, 12, 10, 89, 750),
                new Forecast(R.drawable.sunny, 25, 5, 73, 771),
                new Forecast(R.drawable.sun, 18, 4, 85, 769),
                new Forecast(R.drawable.snow, -3, 1, 59, 741),
                new Forecast(R.drawable.cloud2, 7, 2, 51, 749),
                new Forecast(R.drawable.cloud1, 11, 6, 64, 742)
        );
    }

    class Forecast implements Serializable {

        private int imgId;
        protected int temperature;
        private float wind;
        private int humidity;
        private int pressureMm;
        private long timeStamp;

        public Forecast(int imgId, int temperature, float wind, int humidity, int pressureMm) {
            this.imgId = imgId;
            this.temperature = temperature;
            this.wind = Math.abs(wind);
            this.humidity = Math.abs(humidity);
            this.pressureMm = Math.abs(pressureMm);
            this.timeStamp = new Date().getTime();
        }

        protected int mmToMb(int mm) {
            return (int)(mm * 1.333f);
        }
    }
}

File: kApp/src/main/java/k/s/yarlykov/data/provider/ForecastProvider.kt
package k.s.yarlykov.data.provider

import k.s.yarlykov.R
import java.io.Serializable

class Forecast (
        val imgId: Int,
        internal var temperature: Int,
        internal var wind: Float,
        internal var humidity: Int,
        internal var pressureMm: Int) : Serializable {

    fun mmToMb(mm: Int) = (mm * 1.333f).toInt()
    fun mbToMm(mb: Int) = (mb * 0.75006f).toInt()
}

object ForecastProvider {

    private val forecasts = listOf(
            Forecast(R.drawable.rain, 12, 10f, 89, 750),
            Forecast(R.drawable.sunny, 25, 5f, 73, 771),
            Forecast(R.drawable.sun, 18, 4f, 85, 769),
            Forecast(R.drawable.snow, -3, 1f, 59, 741),
            Forecast(R.drawable.cloud2, 7, 2f, 51, 749),
            Forecast(R.drawable.cloud1, 11, 6f, 64, 742)
    )

    // По индексу из списка
    fun getForecastByIndex(num: Int): Forecast {
        var index = Math.abs(num)
        if (index >= forecasts.size) index = forecasts.size % index
        return forecasts[index]
    }
}